### PR TITLE
Choose sync peer other than leader when possible

### PR DESCRIPTION
**Pull request by danieldoglas**
### Details

This PR changes the logic in `_updateSyncPeer` to choose the leader as a sync peer only if no other nodes are available. The change should help reduce processing pressure on the leader node.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/455682

### Tests

_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

File: sqlitecluster/SQLiteNode.cpp
Comment (iwiznia) [inline sqlitecluster/SQLiteNode.cpp:1282]: Is this change supposed to be in this PR? Seems kind of unrelated to the other change.

**Review by iwiznia — DISMISSED** [review]

File: sqlitecluster/SQLiteNode.cpp
Comment (cead22) [reply sqlitecluster/SQLiteNode.cpp:1282]: I think this PR should be on hold til the other one is merged

**Comment by cead22** [comment]
Change looks good. Happy to approve when the other PR is merged and this one only has the diff for this change

**Comment by danieldoglas** [comment]
Changed the bae PR so it only shows the needed changes

File: sqlitecluster/SQLiteNode.cpp
Comment (danieldoglas) [reply sqlitecluster/SQLiteNode.cpp]: this was the other one: https://github.com/Expensify/Bedrock/pull/2039

I changed the base branch to that PR so you can see only the necessary changes.

**Comment by danieldoglas** [comment]
@cead22 @iwiznia @iwiznia all yours to review.

**Review by iwiznia — DISMISSED** [review]

**Comment by iwiznia** [comment]
Approved (though not showing for me in the right pane), anyway leaving it for @cead22 and @tylerkaraszewski to review too

**Review by cead22 — DISMISSED** [review]

File: sqlitecluster/SQLiteNode.cpp
Comment (tylerkaraszewski) [inline sqlitecluster/SQLiteNode.cpp:2219]: I don't think we need this (see below), we can just `continue` without setting `isLeaderValidPeer`.

File: sqlitecluster/SQLiteNode.cpp
Comment (tylerkaraszewski) [inline sqlitecluster/SQLiteNode.cpp:2247]: I don't think we need `isLeaderValidPeer`, can we just use this here?
```
if (!newSyncPeer && _leadPeer) {
    newSyncPeer = _leadPeer;
} 
```

**Review by tylerkaraszewski — CHANGES_REQUESTED** [review]

File: sqlitecluster/SQLiteNode.cpp
Comment (danieldoglas) [reply sqlitecluster/SQLiteNode.cpp]: Done

File: sqlitecluster/SQLiteNode.cpp
Comment (danieldoglas) [reply sqlitecluster/SQLiteNode.cpp]: Done

**Review by cead22 — APPROVED** [review]

**Review by tylerkaraszewski — APPROVED** [review]

**Pull request merged**
